iModelCore/ecobjects/PublicApi/ECObjects/ECSchema.h

```diff
@@ -3987,7 +3987,13 @@ friend struct SchemaJsonWriter;
     //! @param[in]    thisSchema                   The base schema to check the references of
     //! @param[in]    potentiallyReferencedSchema  The schema to search for
     //! @return True if thatSchema is referenced by thisSchema, false otherwise
-    ECOBJECTS_EXPORT static bool IsSchemaReferenced (ECSchemaCR thisSchema, ECSchemaCR potentiallyReferencedSchema, ECN::SchemaMatchType matchType);
+    ECOBJECTS_EXPORT static bool IsSchemaReferenced (ECSchemaCR thisSchema, ECSchemaCR potentiallyReferencedSchema, ECN::SchemaMatchType matchType, bool checkIndirectReferences = false);

```

Comment:
Do not complicate the IsScheamReferenced method with a check to indirect references.  I'm not sure why we need this check to solve the current problem.

Reply:
Removed `checkIndirectReferences` optional parameter